pkg/apis/networking/v1/defaults_test.go

```diff
@@ -235,7 +239,68 @@ func TestSetDefaultNetworkPolicy(t *testing.T) {
 	}
 }
 
+func TestSetDefaultIngress(t *testing.T) {
```

Comment:
I assume that if the Class is set explicitly without the usage of the annotation, then we should get a valid IngressClass. Does this sound correct? If so can add a test?

Reply:
Not sure I'm understanding this. I added a test that might cover the use case you're talking about - ensuring a ingress class is not interfered with when no annotation is present. Alternatively are you suggesting that a corresponding IngressClass resource should exist if class is set on an Ingress? I had thought of IngressClass as an optional way to add additional configuration for a given Ingress class.